lib/utils/sync_map.go

```diff
@@ -78,16 +78,75 @@ func (s *SyncMap[K, V]) LoadAndDelete(key K) (value V, ok bool) {
 	return value, ok
 }
 
-// Range calls a function sequentially for each key and value in the map. Note
-// that the map is not locked between evaluations of f.
+// Range calls a function sequentially for each key and value in the map.
+// Caution: The map is ony locked while creating a copy of the map values to
+// iterate over; it is *not* locked during the actual iteration over that copy,
+// nor while f is being evaluated.
 func (s *SyncMap[K, V]) Range(f func(key K, value V) bool) {
-	s.mu.RLock()
-	items := maps.Clone(s.values)
-	s.mu.RUnlock()
+	items := s.Clone()
 
 	for key, value := range items {
 		if !f(key, value) {
 			return
 		}
 	}
 }
+
+// Clear clears the underlying map
+func (s *SyncMap[K, V]) Clear() {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	if s.values != nil {
+		clear(s.values)
+	}
+}
+
+// Set takes ownership of a map, replacing the existing managed map with the
+// supplied value
+func (s *SyncMap[K, V]) Set(m map[K]V) {
```

Comment:
I wonder if this use case would be better served by a constructor.

Also, I understand that your inspiration for some of these utilities is Rust, but the reality is Go does not have Rust's ownership semantics and there is no such thing as "taking ownership of a map" in Go. The caller of this method has just as much ownership over `m` before and after this method runs.

Reply:
I see where you're coming from, but there are cases in Okta Reconciler clients that currently wants to entirely replace the map under management post-construction. 

And perhaps "ownership" is not the right word - it certainly has overtones from Rust that are not directly applicable to Go. I'm just trying to convey the idea that all bets are off if you directly modify `m` outside after calling this. Maybe just  spelling it out is better.